From d09b32e7b4c1dcc5919ef757a0e20ed29135e181 Mon Sep 17 00:00:00 2001
From: Kurt Lee <kurt.lee@cypress.com>
Date: Tue, 18 Jan 2022 21:57:37 -0600
Subject: [PATCH 114/179] brcmfmac: Improve the delay during scan

wpa_supplicant sends NL80211_CMD_GET_SURVEY command every after normal
scan, which causes brcmfmac surveying each channel again.

Currently brcmfmac implments brcmf_cfg80211_dump_survey focusing on AP
mode, thus this change will return the request other than AP mode.

Signed-off-by: Kurt Lee <kurt.lee@infineon.com>

---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
index 63bdb5115237..9bc23fbc3c92 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
@@ -8491,6 +8491,9 @@ brcmf_cfg80211_dump_survey(struct wiphy *wiphy, struct net_device *ndev,
 
 	brcmf_dbg(TRACE, "Enter: channel idx=%d\n", idx);
 
+	if (!brcmf_is_apmode(ifp->vif))
+		return -ENOENT;
+
 	/* Do not run survey when VIF in CONNECTING / CONNECTED states */
 	if ((test_bit(BRCMF_VIF_STATUS_CONNECTING, &ifp->vif->sme_state)) ||
 	    (test_bit(BRCMF_VIF_STATUS_CONNECTED, &ifp->vif->sme_state))) {
-- 
2.17.1

